felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -625,6 +645,35 @@ func (m *bpfEndpointManager) onProfileRemove(msg *proto.ActiveProfileRemove) {
 	m.markEndpointsDirty(m.profilesToWorkloads[profID], "profile")
 	delete(m.profiles, profID)
 	delete(m.profilesToWorkloads, profID)
+	if m.bpfPolicyDebugEnabled {
+		m.polNameToRuleIDs[profID.Name] = markAllRuleIdsDirty(m.polNameToRuleIDs[profID.Name])
+	}
+}
+
+func (m *bpfEndpointManager) removeDirtyPolicies() {
+	for k, v := range m.polNameToRuleIDs {
```

Comment:
rather than iterate over all policies, wouldn't it be easier to put the dirty ones in a set then iterate over that set? If there are many policies but only few/none dirty, we do a lot of work for nothing.